Lazy Load Scatterplot Polygons

ilan-gold opened the pull request:
This allows [the million cell satija demo](http://localhost:3000/?url=data%3A%2C%7B+++%22version%22%3A+%221.0.1%22%2C+++%22name%22%3A+%22Covid+19+Cell+Atlas+Bronchoalveolar+Lavage+Fluid%22%2C+++%22description%22%3A+%22Dataset+from+https%3A%2F%2Fwww.covid19cellatlas.org%2Findex.patient.html+visualized+at+https%3A%2F%2Fcellxgene.cziscience.com%2Fe%2Fae466876-e313-45ca-9e12-88e41b0edf89.cxg%2F%22%2C+++%22datasets%22%3A+%5B+++++%7B+++++++%22uid%22%3A+%2240334a99-2be7-4904-bc75-102609fe53e9%22%2C+++++++%22name%22%3A+%2240334a99-2be7-4904-bc75-102609fe53e9%22%2C+++++++%22files%22%3A+%5B+++++++++%7B+++++++++++%22type%22%3A+%22cells%22%2C+++++++++++%22fileType%22%3A+%22anndata-cells.zarr%22%2C+++++++++++%22url%22%3A+%22https%3A%2F%2Fvitessce-demo-data.storage.googleapis.com%2Fcxg_examples%2Flung_mapped.zarr%22%2C+++++++++++%22options%22%3A+%7B+++++++++++++%22mappings%22%3A+%7B+%22UMAP%22%3A+%7B+%22key%22%3A+%22obsm%2FX_umap%22%2C+%22dims%22%3A+%5B0%2C+1%5D+%7D+%7D+++++++++++%7D+++++++++%7D%2C+++++++++%7B+++++++++++%22type%22%3A+%22cell-sets%22%2C+++++++++++%22fileType%22%3A+%22anndata-cell-sets.zarr%22%2C+++++++++++%22url%22%3A+%22https%3A%2F%2Fvitessce-demo-data.storage.googleapis.com%2Fcxg_examples%2Flung_mapped.zarr%22%2C+++++++++++%22options%22%3A+%5B+++++++++++++%7B+++++++++++++++%22groupName%22%3A+%22Annotation+Level+1%22%2C+++++++++++++++%22setName%22%3A+%22obs%2Fannotation.l1%22+++++++++++++%7D%2C+++++++++++++%7B+++++++++++++++%22groupName%22%3A+%22Annotation+Level+2%22%2C+++++++++++++++%22setName%22%3A+%22obs%2Fannotation.l2%22+++++++++++++%7D%2C+++++++++++++%7B+++++++++++++++%22groupName%22%3A+%22Diagnosis%22%2C+++++++++++++++%22setName%22%3A+%22obs%2FDiagnosis%22+++++++++++++%7D%2C+++++++++++++%7B+++++++++++++++%22groupName%22%3A+%22Disease%22%2C+++++++++++++++%22setName%22%3A+%22obs%2FDisease%22+++++++++++++%7D%2C+++++++++++++%7B+++++++++++++++%22groupName%22%3A+%22Sex%22%2C+++++++++++++++%22setName%22%3A+%22obs%2FSex%22+++++++++++++%7D+++++++++++%5D+++++++++%7D%2C+++++++++%7B+++++++++++%22type%22%3A+%22expression-matrix%22%2C+++++++++++%22fileType%22%3A+%22anndata-expression-matrix.zarr%22%2C+++++++++++%22url%22%3A+%22https%3A%2F%2Fvitessce-demo-data.storage.googleapis.com%2Fcxg_examples%2Flung_mapped.zarr%22%2C+++++++++++%22options%22%3A+%7B+%22matrix%22%3A+%22X%22+%7D+++++++++%7D+++++++%5D+++++%7D+++%5D%2C+++%22initStrategy%22%3A+%22auto%22%2C+++%22coordinationSpace%22%3A+%7B+%22embeddingType%22%3A+%7B+%22UMAP%22%3A+%22UMAP%22+%7D+%7D%2C+++%22layout%22%3A+%5B+++++%7B+++++++%22component%22%3A+%22cellSets%22%2C+++++++%22h%22%3A+3%2C+++++++%22w%22%3A+3%2C+++++++%22x%22%3A+9%2C+++++++%22y%22%3A+0%2C+++++++%22coordinationScopes%22%3A+%7B%7D+++++%7D%2C+++++%7B+++++++%22component%22%3A+%22genes%22%2C+++++++%22h%22%3A+3%2C+++++++%22w%22%3A+3%2C+++++++%22x%22%3A+9%2C+++++++%22y%22%3A+3%2C+++++++%22coordinationScopes%22%3A+%7B%7D+++++%7D%2C+++++%7B+++++++%22component%22%3A+%22scatterplot%22%2C+++++++%22h%22%3A+6%2C+++++++%22w%22%3A+9%2C+++++++%22x%22%3A+0%2C+++++++%22y%22%3A+0%2C+++++++%22coordinationScopes%22%3A+%7B+%22embeddingType%22%3A+%22UMAP%22+%7D+++++%7D+++%5D+%7D&theme=dark) to have a faster initial load time.

keller-mark commented:
Just made this PR https://github.com/vitessce/vitessce/pull/962

keller-mark commented on src/components/scatterplot/ScatterplotSubscriber.js:
```diff
@@ -150,13 +151,26 @@ export default function ScatterplotSubscriber(props) {
+      setCellSetPolygonCache(cache => [...cache, [cellSetSelection, newCellSetPolygons]]);
+      return newCellSetPolygons;
+    }
+    return cacheGet(cellSetPolygonCache, cellSetSelection);
```
```suggestion
    return cacheGet(cellSetPolygonCache, cellSetSelection) || [];
```

The http://localhost:3000/?dataset=dries-2019 demo crashes when `cellSetPolygons` is undefined

ilan-gold commented:
Seems like the cluster labels also disappeared....

ilan-gold commented:
@keller-mark I added some checks to make sure the things that are needed are present before calculating the polygons.  I also added a check for whether or not the labels are visible since the labels seem to rely on the polygons.
